src/main.js

```diff
@@ -993,6 +993,19 @@ class MiniGraphCard extends LitElement {
   getEndDate() {
     const date = new Date();
     switch (this.config.group_by) {
+      case 'year':
+        date.setDate(1); // First day
+        date.setMonth(0); // of January
+        date.setHours(0, 0, 0); // 00:00:00
+        break;
+      case 'month':
+        date.setDate(1); // First day of current month
+        date.setHours(0, 0, 0);
+        break;
+      case 'week':
+        date.setDate(date.getDate() - date.getDay() + 1); // First day of the week
```

Comment:
I just flipped through the PRs here and noticed: should this not be the first day of *next* week at 0h00? See the date case below... The same applies to month and year.